saplings/circuits/src/components/circuitsTable/TableHeader.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2018-2020 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { Circuit } from '../../data/processCircuits';
+
+const sortCircuits = (circuits, action) => {
```

Comment:
I don't know. I think it makes sense more in the header, because the circuits table, and the header in particular are the only components that have knowledge of what columns are available for sorting. For me it does not make sense to leak that knowledge to the circuits reducer that is defined in the parent.

Reply:
I see your point. However, since we are using that reducer to manage the state of our circuits, I would say that it does have knowledge of what properties each circuit has. There isn't anything wrong with crafting that reducer in a way that suits our needs. 

When we are dispatching an action from a view, we are telling it to update the state in some way based on the keyword.  We also have the option to include some other data in the action's payload, but I think that requiring the view component to provide the logic for updating state would be too far. That should be the responsibility of the reducer.

If you move this functionality (and possibly the filter reducer) to the Content component, or even a different module, I think that would improve the separation of concerns in this part of the app.

Reply:
Okay, I see your point. I moved everything related to managing circuits state to its own file